arkouda/dataframe.py

```diff
@@ -1802,6 +1802,49 @@ def isin(self, values: Union[pdarray, Dict, Series, DataFrame]) -> DataFrame:
 
         return DataFrame(df_def, index=self.index)
 
+    def corr(self) -> DataFrame:
+        """
+        Return new DataFrame with pairwise correlation of columns
+
+        Returns
+        -------
+        DataFrame
+            Arkouda DataFrame containing correlation matrix of all columns
+
+        Raises
+        ------
+        RuntimeError
+            Raised if there's a server-side error thrown
+
+        See Also
+        --------
+        pdarray.corr
+
+        Notes
+        -----
+        Generates the correlation matrix using Pearson R for all columns
+
+        Attempts to convert to numeric values where possible for inclusion in the matrix.
+        """
+        # TODO converting Strings to Categorical goes out of scope, i don't love registering it
```

Comment:
Can we remove this and add an issue for looking into this further? I am not a fan of pushing `TODO`s in master.

Reply:
Removed! And while I was writing up the issue, I actually came up with a workaround, if u don't mind checking it out and letting me know what u think

```python
>>> s = ak.array([f"string {i}" for i in range(10)])
>>> c = ak.Categorical(s)
>>> s2 = ak.array([f"{i} string {i}" for i in range(10)])
>>> c2 = ak.Categorical(ak.array([f'my_cat {i}' for i in range(10)]))
>>> ak.DataFrame({"s": s, "c": c, "s2": s2, "c2": c2}).corr()
          s         c        s2        c2
0  1.000000  1.000000 -0.127273  0.090909
1  1.000000  1.000000 -0.127273  0.090909
2 -0.127273 -0.127273  1.000000  0.272727
3  0.090909  0.090909  0.272727  1.000000 (4 rows x 4 columns)
```

Reply:
@pierce314159 if this work around is functional I would much prefer using that over having to register and deregister things. Though that is a matter of preference on my part. I think this looks simple enough as well that we could update this as part of the current PR and close the subsequent issue you created. Unless you believe there is still a better way to do this. 

*Leaving this thread open for now. The original concern was addressed., but I think there is more discussion to have at this time.*

Reply:
I also wanted to note, I am not currently seeing the register/unregister code here anymore, did you remove it in anticipation of making the above update?

Reply:
Oh no the code you're seeing is the fix for this. Previously we had
```python
args = {
    "size": len(self.columns),
    "columns": self.columns,
    "data_names": [numeric_help(self[c]).name for c in self.columns],
}
```
which would go out of scope because `Categorical`s created by `numeric_help` go out of scope.

To fix this i created a list to store the `Categorical`s created by `numeric_help` which is gonna stick around for a bit
```python
numeric_cols = [numeric_help(self[c]) for c in self.columns]
args = {
    "size": len(self.columns),
    "columns": self.columns,
    "data_names": [col.name for col in numeric_cols],
}
```
So this takes care of the need to register/unregister

Once your PR #1662, is merged this will just be
```python
numeric_cols = [numeric_help(self[c]) for c in self.columns]
args = {
    "size": len(self.columns),
    "columns": self.columns,
    "data_names": numeric_cols,
}
```
which is why i call this a workaround cause it feels a little silly to create a var for this when it's only used once... but it's wayyyyy better than the reg/unreg stuff

Reply:
actually, after merging 1662, the way we want to do this just works (i'm guessing because we're storing the actual pdarray in `args` now and not just the name?). So this has been updated to 
```python
args = {
    "size": len(self.columns),
    "columns": self.columns,
    "data_names": [numeric_help(self[c]) for c in self.columns],
}
```